src/raftstore/store/snap/util.rs

```diff
@@ -0,0 +1,292 @@
+// Copyright 2018 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+#![allow(dead_code)] // TODO: remove this.
+use std::path::PathBuf;
+use std::sync::atomic::{AtomicBool, AtomicU64, Ordering};
+use std::sync::Arc;
+use std::{error, fmt, io};
+
+use kvproto::raft_serverpb::{RegionLocalState, SnapshotMeta};
+use rocksdb::DB;
+
+use storage::{CfName, CF_DEFAULT, CF_LOCK, CF_WRITE};
+use util::codec;
+
+pub const SNAPSHOT_VERSION: u64 = 2;
+pub const SNAPSHOT_CFS: &[CfName] = &[CF_DEFAULT, CF_LOCK, CF_WRITE];
+pub const SNAP_GEN_PREFIX: &str = "gen";
+pub const SNAP_REV_PREFIX: &str = "rev";
+pub const META_FILE_NAME: &str = "meta";
+pub const SST_FILE_SUFFIX: &str = ".sst";
+pub const TMP_FILE_SUFFIX: &str = ".tmp";
+pub const CLONE_FILE_SUFFIX: &str = ".clone";
+
+quick_error! {
```

Comment:
I'd like to put these error under `snap::Error`.

Reply:
After the reader and the writer is used in SnapManager, the `Error` will replace `snap::Error`. So let's keep it in this PR.